### PR TITLE
Allow overwriting existing index template

### PR DESCRIPTION
**Pull request by chancez**
Basically the same as https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-template_overwrite

### CODE REVIEW

**Comment by chancez** [comment]
I'd be glad to write a test for this, I just need a bit of guidance, it doesn't seem like existing tests verify that the old behavior didn't update the template if it already exists, which I'd need to add, and then another test to validate it is updated when the overwrite setting is enabled.

**Comment by chancez** [comment]
Huzzah. Got test coverage back up for the new functionality, and improved the existing tests to assert the existing behavior as well. Let me know if you think I'm missing some tests.

**Comment by chancez** [comment]
@repeatedly Are you able to review this or does @pitr the only person who can merge/review things for this project?

**Comment by repeatedly** [comment]
@chancez I can't review because I'm not familiar with ES itself.
AFAIK, @pitr and @dterror are main maintainers but I'm not sure how they have a time to review all patches.
If they can, adding one more maintainer who knows ruby and fluentd is one approach.

File: lib/fluent/plugin/out_elasticsearch.rb [hunk 1]
```diff
@@ -55,6 +55,7 @@ class ConnectionFailure < StandardError; end
   config_param :flatten_hashes_separator, :string, :default => "_"
   config_param :template_name, :string, :default => nil
   config_param :template_file, :string, :default => nil
+  config_param :template_overwrite, :string, :default => false
```
Comment (pitr) [inline lib/fluent/plugin/out_elasticsearch.rb]: should be `:bool`

Comment (chancez) [inline lib/fluent/plugin/out_elasticsearch.rb]: Ah, good catch. Fixed.

**Comment by chancez** [comment]
@pitr Yep. That is the expected behavior, and also what logstash does. Generally there will be no conflicts and overwriting at boot up is fine since most the time the template will be the same across all instances of fluentd.

**Comment by chancez** [comment]
@pitr Can we merge this? Anything outstanding?

**Comment by dannyk81** [comment]
@pitr @cosmo0920 just wondering where this stands?

**Comment by chancez** [comment]
I just rebased. It should be good to go.

**Comment by cosmo0920** [comment]
I'd appreciate if you sent back port PR into v0.12 branch.
Thanks!

**Comment by chancez** [comment]
@cosmo0920 as a separate PR?

**Comment by cosmo0920** [comment]
@chancez Yes. Could you send a separate PR into v0.12 branch?

**Comment by cosmo0920** [comment]
I'm planning to merge and release this PR as v2.3.0.
And then, I'm also planning to backport and release as v1.13.0.

**Comment by cosmo0920** [comment]
I've release this as v2.3.0.
Thank you for your contribution, @chancez !

**Comment by chancez** [comment]
Thanks @cosmo0920.

**Pull request merged**